DemoApp/ChatPresenter.swift

```diff
@@ -13,16 +13,8 @@ extension UIViewController {
         
         // Create client
         let config = ChatClientConfig(apiKey: .init(userCredentials.apiKey))
-        let client = ChatClient(config: config)
-        
-        // Log in the current user
-        let currentUserController = client.currentUserController()
-        currentUserController.setUser(
-            userId: userCredentials.id,
-            name: userCredentials.name,
-            imageURL: userCredentials.avatarURL,
-            token: userCredentials.token
-        ) { error in
+        let client = ChatClient(config: config, tokenProvider: .hardCoded(userCredentials.token))
+        client.reconnectIfNeeded { error in
```

Comment:
Why don't we just `connect` since this is the initial connection part?

Reply:
There's no initial connection guys, it happens implicitly ⚠️ This name should be `reloadUserIfNeeded` or something like that. Managing connection manually is an advanced topic, we shouldn't even use anything related to that in naming, so it's not confusing.

Reply:
Currently `connect/disconnect` API is occupied for web-socket actions but we can use it here for sure.

Named it `reconnectIfNeeded` to highight it can be safely called for a couple of times and none unnecessary job will be done (e.g. re-establish the connection) if the current user is not changed.

It is also expected to be used by **passive** clients but but `passiveClient.connect` feels too active 😄